### PR TITLE
Add OCP 3.11 scale run environment details

### LINKED ISSUE #20868: Tested Maximum page should be more specific
### Which section(s) is the issue in?
https://github.com/openshift/openshift-docs/blob/master/modules/openshift-cluster-maximums.adoc

### What needs fixing?
openshift-cluster-maximums.adoc Should become `openshift-<platform>-cluster-maximums.adoc` and contain how did we achieve a specific scale for that specific platform. 

Today our documentation only discusses what we achieved, how what we achieved things on.

### PR DESCRIPTION
**Pull request by chaitanyaenr**
This commit:
- Adds details about the environment including cloud platform used
  for running the performance and scale tests on OCP 3.11 as well as
  info about the cluster configuration like instance types, CPU,
  memory, disk, node count e.t.c.
- Adds info on why we went with particular configuration as well as
  pointers to the docs wherever required.

Fixes #20868

### CODE REVIEW

**Comment by jtaleric** [comment]
@chaitanyaenr add to the commit message : `fixes #20868`

**Comment by jtaleric** [comment]
LGTM!

**Comment by ahardin-rh** [comment]
LGTM. @mffiedler Can you please provide a QE review? Thanks!

File: scaling_performance/cluster_maximums.adoc [hunk 1]
```diff
@@ -102,6 +102,66 @@ application requirements.]
...
+| 1
+
+| Container Native Storage footnoteref:[cns, Container Native Storage or Ceph storage nodes are backed by NVMe disks.]
+| 16
+| 65280
+| 200
+| Yes, NVMe
+| 3
+
+| Jump Node footnoteref:[jumpnode, Jump node is part of the OCP network and is used to orchestrate the performance and scale tests.]
```
Comment (mffiedler) [inline scaling_performance/cluster_maximums.adoc:149]: Bastion Node would be better terminology and more familiar to the audience.

Bastion Node footnoteref:[bastionnode, The Bastion node is part of the OCP network and is used to orchestrate the performance and scale tests.]

Comment (chaitanyaenr) [reply scaling_performance/cluster_maximums.adoc]: Fixed it, thanks.

**Comment by ahardin-rh** [comment]
Thank you!

**Pull request merged**